src/kbmod/search/image_kernels.cu

```diff
@@ -347,74 +352,109 @@ __global__ void device_get_coadd_stamp(int num_images, int width, int height, fl
 
         // Predict the trajectory's position including the barycentric correction if needed.
         float cTime = image_data.imageTimes[t];
-        int currentX = trj.x + int(trj.xVel * cTime);
-        int currentY = trj.y + int(trj.yVel * cTime);
+        int currentX = int(trj.x + trj.xVel * cTime);
+        int currentY = int(trj.y + trj.yVel * cTime);
         if (image_data.baryCorrs != nullptr) {
             baryCorrection bc = image_data.baryCorrs[t];
             currentX = int(trj.x + trj.xVel*cTime + bc.dx + trj.x*bc.dxdx + trj.y*bc.dxdy);
             currentY = int(trj.y + trj.yVel*cTime + bc.dy + trj.x*bc.dydx + trj.y*bc.dydy);
         }
 
-        // Get the stamp and add it to the running totals..
-        for (int stamp_y = 0; stamp_y < stamp_width; ++stamp_y) {
-            int img_y = currentY - params.radius + stamp_y;
-            for (int stamp_x = 0; stamp_x < stamp_width; ++stamp_x) {
-                int img_x = currentX - params.radius + stamp_x;
-                if ((img_x >= 0) && (img_x < width) && (img_y >= 0) && (img_y < height)) {
-                    int pixel_index = pixels_per_image * t + img_y * width + img_x;
-                    if (image_vect[pixel_index] != NO_DATA) {
-                        int stamp_index = stamp_y * stamp_width + stamp_x;
-                        sum[stamp_index] += image_vect[pixel_index];
-                        count[stamp_index] += 1.0;
-                    }
-                }
+        // Get the stamp and add it to the list of values.
+        int img_y = currentY - params.radius + stamp_y;
+        int img_x = currentX - params.radius + stamp_x;
+        if ((img_x >= 0) && (img_x < width) && (img_y >= 0) && (img_y < height)) {
+            int pixel_index = width * height * t + img_y * width + img_x;
+            if (image_vect[pixel_index] != NO_DATA) {
+                values[num_values] = image_vect[pixel_index];
+                ++num_values;
             }
-        }    
+        }
     }
 
-    // Compute the mean if needed.
-    if (params.stamp_type == STAMP_MEAN) {
-        for (int i = 0; i < stamp_ppi; ++i) {
-            sum[i] = (count[i] > 0.0) ? sum[i]/count[i] : 0.0;
+    // Do the actual computation from the values.
+    float result = 0.0;
+    if ((params.stamp_type == STAMP_MEDIAN) && (num_values > 0)) {
```

Comment:
Since the following ~20-30 lines don't many any sense if `num_values == 0`, maybe just go ahead and return if that's the case (or set results[thing] == 0). I think it makes the flow read more naturally. Or you could turn the remainder into a proper switch/case clause.

Reply:
Did both (early return and switch/case)